src/functions.cc

```diff
@@ -23,6 +27,77 @@ NAN_METHOD(set_dev_mode) {
   }
 }
 
+static void addTags(v8::Isolate* isolate, const v8::Local<v8::Object>& object,
+                    Tags* tags) {
+  auto context = isolate->GetCurrentContext();
+  auto maybe_props = object->GetOwnPropertyNames(context);
+  if (!maybe_props.IsEmpty()) {
+    auto props = maybe_props.ToLocalChecked();
+    auto n = props->Length();
+    for (uint32_t i = 0; i < n; ++i) {
+      const auto& key = props->Get(i);
+      const auto& value = object->Get(key);
+      const auto& k = std::string(*Nan::Utf8String(key));
+      const auto& v = std::string(*Nan::Utf8String(value));
+      tags->add(k.c_str(), v.c_str());
+    }
+  }
+}
+
+NAN_METHOD(analyze_id) {
+  const char* err_msg = nullptr;
+  std::string name;
+  ValidationIssues res;
+  Tags tags;
+  auto ret = Nan::New<v8::Array>();
+
+  if (info.Length() > 2) {
```

Comment:
As I mentioned in Slack, this seems like it's duplicating the validation code in https://github.com/Netflix-Skunkworks/atlas-node-client/blob/master/src/utils.cc#L94-L123, but I'm OK if this is intentional and we think that there is no risk of divergence. It doesn't have to block this PR.

Reply:
Sounds good. I'm leaning towards using a different approach for writing my metrics related tests, so I think this specific aspect will become less important for my use cases.